CreeDictionary/API/schema.py

```diff
@@ -0,0 +1,58 @@
+"""
+this file contains `TypedDict` classes that effectively serves as json schema for serialized objects
+"""
+from typing import List, Tuple, Union
+from typing_extensions import TypedDict
+
+
+class SerializedDefinition(TypedDict):
+    text: str
+    source_ids: Tuple[str]
+
+
+class SerializedWordform(TypedDict):
+    id: int
+    text: str
+    full_lc: str
+    pos: str
+    analysis: str
+    is_lemma: bool
+    as_is: bool
+    lemma: int  # the id of the lemma
+
+    # ---- calculated properties ---
+    lemma_url: str
+
+    # ---- foreign keys ----
+    definition: List[SerializedDefinition]
+
+
+class SerializedSearchResult(TypedDict):
+    # the text of the match
+    matched_cree: str
+
+    is_lemma: bool
+
+    # ENGLISH | CREE
+    matched_by: str
```

Comment:
In Python 3.8, this could be:

```python
    matched_by: Literal["ENGLISH", "CREE"]
```

...but I guess this doesn't affect our code 🙃

Reply:
![Screenshot from 2020-02-27 13-26-45](https://user-images.githubusercontent.com/44753941/75483824-d302d000-5964-11ea-8f94-ccc3ddb7e3e2.png)

use a real IDE Eddie :wink: